realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -160,15 +159,25 @@ public void onChange(AllJavaTypes object) {
         parent.setFieldObject(child);
         looperThreadRealm.commitTransaction();
 
-        assertEquals(2, looperThreadRealm.where(AllJavaTypes.class).findAll().size());
-        assertEquals(1, counter.get());
+        // This should be on the Looper queue, behind the notification
+        looperThread.postRunnable(
```

Comment:
I have improved the tests based on my current understanding that `beginTransaction` forces the delivery (to the Looper thread) of any outstanding notifications.